### PR TITLE
Make CheckStrictDeps understand goog.declareModuleId

### PR DESCRIPTION
**Pull request by Yannic**
See https://github.com/google/closure-compiler/wiki/Migrating-from-goog.modules-to-ES6-modules

### CODE REVIEW

**Comment by Yannic** [comment]
@alexeagle Can you review?

File: java/com/google/javascript/jscomp/CheckStrictDeps.java [hunk 1]
```diff
@@ -70,7 +70,8 @@ public final void visit(NodeTraversal t, Node n, Node parent) {
       Node parameter = n.getLastChild();
       if (parameter.isString()
           && (callee.matchesQualifiedName("goog.provide")
-              || callee.matchesQualifiedName("goog.module"))) {
+              || callee.matchesQualifiedName("goog.module")
```
[the remark this answers is not in the document]
Comment (Yannic) [reply java/com/google/javascript/jscomp/CheckStrictDeps.java]: As a first step, we could split ClosureWorker into a "Compiler-Worker" and a "Webfiles-Worker" and move the "Compiler-Worker" into the compiler repo (maybe even move `closure_js_{library,binary,test}` into the compiler repo?).

[the remark this answers is not in the document]
Comment (alexeagle) [reply java/com/google/javascript/jscomp/CheckStrictDeps.java]: Worker mode is a particular case where we might want some Java code that wraps the closure compiler main. I'm thinking of domain logic like this PR.
Another way to express it is: wouldn't someone outside of Bazel want CheckStrictDeps to do the right thing? This repo should only have things which are essential to Bazel and not shared with other use cases

[the remark this answers is not in the document]
Comment (Yannic) [reply java/com/google/javascript/jscomp/CheckStrictDeps.java]: I could imagine a use case that, e.g., checks whether `package.json` contains all direct dependencies, but I'm not aware of the existence of such a tool.

The compiler folks already have stated that rules_closure provides a maintenance burden for them (and upgrading to a newer version is always a bit of a pain), so moving some (most) compiler-specific code into the compiler repo sounds like a good goal.

[the remark this answers is not in the document]
Comment (alexeagle) [reply java/com/google/javascript/jscomp/CheckStrictDeps.java]: yes exactly, we are working on a plan to cut scope in rules_closure by just making underlying tools work properly and then call them with a `genrule` or nicer equivalent like https://github.com/bazelbuild/rules_nodejs/tree/master/examples/closure - I'd like to get more of the functionality ported to that model. We'll move tensorboard and maybe a couple projects like js-dossier to that, then rules_closure can either end or be community owned as you and @robfig are mostly doing anyway. Will have some more real update end-of-month I hope.
Anyhow I'll get this PR tested in google3 and try to merge

**Pull request merged**